pkg/daemon/ceph/osd/volume.go

```diff
@@ -149,12 +153,26 @@ func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceO
 		// Reporting
 		reportArgs := append(batchArgs, []string{
 			"--report",
+			"--format",
+			"json",
 		}...)
 
-		if err := context.Executor.ExecuteCommand(false, "", baseCommand, reportArgs...); err != nil {
+		cvOut, err := context.Executor.ExecuteCommandWithOutput(false, "", baseCommand, reportArgs...)
+		if err != nil {
 			return fmt.Errorf("failed ceph-volume report. %+v", err) // fail return here as validation provided by ceph-volume
 		}
 
+		logger.Infof("ceph-volume report: %+v", cvOut)
+
+		var cvReport cephVolReport
+		if err = json.Unmarshal([]byte(cvOut), &cvReport); err != nil {
+			return fmt.Errorf("failed to unmarshal ceph-volume report json. %+v", err)
+		}
+
+		if path.Join("/dev", a.metadataDevice) != cvReport.Vg.Devices {
+			return fmt.Errorf("ceph-volume did not use the expected metadataDevice [%s]", a.metadataDevice)
```

Comment:
This would be skipping OSD creation for all the devices in this batch. Could you add all the devices to the error message?

Reply:
All of the devices will be listed in the human readable c-v output directly above the error in the log.  I can add this, but it would require another variable tracking the devices since they are all just appended to batchArgs.  This could be added in https://github.com/rook/rook/pull/3213 where the osd devices are already available here as a list.

Reply:
true, since it's already in the above output that sounds fine, thanks